torch/_inductor/codegen/triton.py

```diff
@@ -134,18 +134,21 @@ def to_dtype(x, dtype: torch.dtype):
     def to_dtype_bitcast(x, dtype: torch.dtype):
         return f"{x}.to({triton_compute_type(dtype)}, bitcast=True)"
 
-    @staticmethod
-    def constant_val(val):
-        return triton_constant(val)
-
-    @staticmethod
-    def constant(value, dtype):
+    @classmethod
+    def constant(cls, value, dtype):
         if dtype == torch.uint8:
-            tmp = ops.constant_val(value)
-            return ops.to_dtype(tmp, dtype)
-        else:
-            type_ = torch._prims_common.dtype_to_type(dtype)
-            return triton_constant(type_(value))
+            # tl.full is broken for uint8, remove once triton is fixed
+            tmp = cls.constant(value, torch.int16)
+            return cls.to_dtype(tmp, dtype)
+
+        # NOTE: We use a tensor here in order to get the correct type.
+        # Otherwise, e.g. float64 constants would be trunctated to float32.
```

Comment:
We could default to the previous path for `int32` and `float32` constants, to get simpler codegen'd in those cases.